tiny-postgresql-event-store/src/main/kotlin/jp/veka/PostgresClientEventStore.kt

```diff
@@ -0,0 +1,115 @@
+package jp.veka
+
+import jp.veka.converter.ResultSetToEntityMapper
+import jp.veka.executor.QueryExecutor
+import jp.veka.query.QueryBuilder
+import jp.veka.tables.ActiveEventStreamReaderDto
+import jp.veka.tables.EventRecordDto
+import jp.veka.tables.EventRecordTable
+import jp.veka.tables.EventStreamActiveReadersTable
+import jp.veka.tables.EventStreamReadIndexDto
+import jp.veka.tables.SnapshotDto
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import ru.quipy.database.EventStore
+import ru.quipy.domain.ActiveEventStreamReader
+import ru.quipy.domain.EventRecord
+import ru.quipy.domain.EventStreamReadIndex
+import ru.quipy.domain.Snapshot
+import kotlin.reflect.KClass
+
+class PostgresClientEventStore(
+    private val eventStoreSchemaName: String,
+    private val resultSetToEntityMapper: ResultSetToEntityMapper,
+    private val executor: QueryExecutor
+) : EventStore {
+    companion object {
+        val logger: Logger = LoggerFactory.getLogger(PostgresClientEventStore::class.java)
+    }
+    override fun insertEventRecord(aggregateTableName: String, eventRecord: EventRecord) {
+        executor.execute(
+            QueryBuilder.insert(eventStoreSchemaName, EventRecordDto(eventRecord, aggregateTableName))
+        )
+    }
+
+    override fun insertEventRecords(aggregateTableName: String, eventRecords: List<EventRecord>) {
+        executor.execute(
+            QueryBuilder.batchInsert(eventStoreSchemaName, EventRecordTable.name, eventRecords.map { EventRecordDto(it, aggregateTableName) })
+        )
+    }
+
+    override fun tableExists(aggregateTableName: String): Boolean {
+        return true // ???
+    }
+
+    override fun updateSnapshotWithLatestVersion(tableName: String, snapshot: Snapshot) {
+        executor.execute(
+            QueryBuilder.insertOrUpdateQuery(eventStoreSchemaName, SnapshotDto(snapshot, tableName))
```

Comment:
pay more attention on how it's made for mongo. In the presence of concurrent snapshot update, one with a higher version wins.